nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -267,6 +276,22 @@ in
       '';
     };
 
+    enableReload = mkOption {
+      default = true;
+      type = types.bool;
+      description = lib.mdDoc ''
+        Reload Caddy when configuration file changes (instead of restart).
+        The configuration file is exposed at {file}`/etc/${etcConfigFile}`.
+
+        Note that enabling this option requires the [admin API](https://caddyserver.com/docs/caddyfile/options#admin)
+        to not be turned off.
+
+        If you enable this option, consider setting [`grace_period`](https://caddyserver.com/docs/caddyfile/options#grace-period)
```

Comment:
We should do this by default

Reply:
You mean setting `grace_period`?

There is a reason why Caddy does not set this by default.
I don't think this module needs to be made even more opinionated. It already deviates a quite a bit from upstream.

I think the note suffices.

Reply:
> You mean setting `grace_period`?

Maybe we don't. I thought this was a no brainer setting but actually this seems to be a worse implementation of nginx reload feature which does not block new connections.

> It already deviates a quite a bit from upstream.

That is not necessarily bad if the program is more secure or the option should almost always be turned on, which I don't think is the case here.